editor/src/components/inspector/common/longhand-shorthand-hooks.ts

```diff
@@ -0,0 +1,236 @@
+  ShorthandKey extends ParsedPropertiesKeys
+>(
+  longhand: LonghandKey,
+  shorthand: ShorthandKey,
+  longhandPropertyStatus: PropertyStatus,
+  shorthandPropertyStatus: PropertyStatus,
+  longhandValue: ParsedProperties[LonghandKey],
+  shorthandValue: ParsedProperties[ShorthandKey],
+  orderedPropKeys: (LonghandKey | ShorthandKey)[][], // multiselect
+): { value: ParsedProperties[LonghandKey]; propertyStatus: PropertyStatus } {
+  const allPropKeysEqual = orderedPropKeys.every((propKeys) => {
+    return arrayEquals(propKeys, orderedPropKeys[0])
+  })
+
+  const propKeys = orderedPropKeys[0] ?? []
+  const lastKey = last(propKeys)
+  if (lastKey == null) {
+    return {
+      value: longhandValue,
+      propertyStatus: longhandPropertyStatus,
+    }
+  } else {
+    if (lastKey === longhand) {
+      return {
+        value: longhandValue,
+        propertyStatus: allPropKeysEqual
+          ? longhandPropertyStatus
+          : {
+              ...longhandPropertyStatus,
+              set: true,
+              controlled: true,
+              overwritable: false,
+              identical: false,
+            },
+      }
+    } else {
+      // Important: we ASSUME that the transformed shorthand value is an object,
+      // where the keys are the longhand keys and the values are the individual longhand values
+      if (longhand in shorthandValue) {
+        return {
+          value: (shorthandValue as any)?.[longhand] as ParsedProperties[LonghandKey],
+          propertyStatus: allPropKeysEqual
+            ? shorthandPropertyStatus
+            : {
+                ...shorthandPropertyStatus,
+                set: true,
+                controlled: true,
+                overwritable: false,
+                identical: false,
+              },
+        }
+      } else {
+        throw new Error(
+          `We didn't find longhand key ${longhand} in resolved shorthand value for shorthand ${shorthand}`,
+        )
+      }
+    }
+  }
+}
+
+export function useInspectorInfoLonghandShorthand<
+  LonghandKey extends ParsedPropertiesKeys,
+  ShorthandKey extends ParsedPropertiesKeys
+>(
+  longhand: LonghandKey,
+  shorthand: ShorthandKey,
+  pathMappingFn: PathMappingFn<LonghandKey | ShorthandKey>,
+): Omit<InspectorInfo<ParsedProperties[LonghandKey]>, 'useSubmitValueFactory'> & {
+  orderedPropKeys: Array<Array<LonghandKey | ShorthandKey>>
+} {
+  const dispatch = useEditorState(
+    (store) => store.dispatch,
+    'useInspectorInfoLonghandShorthand dispatch',
+  )
+  const inspectorTargetPath = useKeepReferenceEqualityIfPossible(
+    useContextSelector(InspectorPropsContext, (contextData) => contextData.targetPath, deepEqual),
+  )
+  const { selectedViewsRef } = useInspectorContext()
+  const orderedPropKeys = useGetOrderedPropertyKeys(pathMappingFn, [longhand, shorthand])
+  const longhandInfo = useInspectorInfo(
+    [longhand],
+    (v) => v[longhand],
+    () => {
+      return null as any
```

Comment:
this is fine because we don't use `useInspectorInfo`'s builtin `onSubmitValue`, which means we don't need this untransformValue neither

Reply:
Can you make this a comment please

Reply:
yeah sure!